server/game/cards/19.1-AS01/Crab/ProtectedMerchant.js

```diff
@@ -1,17 +1,27 @@
 const DrawCard = require('../../../drawcard.js');
-const { Phases } = require('../../../Constants');
+const { CardTypes } = require('../../../Constants');
 const AbilityDsl = require('../../../abilitydsl');
 
 class ProtectedMerchant extends DrawCard {
     setupCardAbilities() {
-        this.reaction({
-            title: 'Put 1 fate on this character',
-            when: {
-                onCharacterEntersPlay: (event, context) => event.card === context.source && context.game.currentPhase === Phases.Dynasty
-            },
-            gameAction: AbilityDsl.actions.placeFate()
+        this.persistentEffect({
+            effect: AbilityDsl.effects.modifyGlory(() =>
+                Math.min(2, this.getHoldingsInPlay())
+            )
         });
     }
+
+    getHoldingsInPlay() {
+        return this.game.allCards.reduce(
+            (sum, card) =>
+                card.type === CardTypes.Holding &&
+                card.isFaceup() &&
+                card.isInProvince()
+                    ? sum + 1
+                    : sum,
+            0
```

Comment:
From the card text, this is incorrect: `for each holding you control` this is checking all holdings